File: tests/nodejs/cases/case_064_enrich_shipment.js
/**
 * Task case 064: enrich_shipment
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case064EnrichShipment(input) {
  // Task: enrich shipment records and prepare transport-ready payload.
  const taskName = 'enrich_shipment';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'shipment',
    stage: 'enrich',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is nested inside retained envelope persisted to retained audit state.
  const envelope = { kind: 'audit-envelope', payload };
  retainedAudit.push(envelope);
  return 'ok';
}

module.exports = {
  case064EnrichShipment,
};

File: tests/python/cases/case_064_enrich_shipment.py
"""
Task case 064: enrich shipment payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_064_enrich_shipment(input_data):
    # Task: enrich shipment records and prepare transport-ready payload.
    task_name = 'enrich_shipment'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'shipment',
        'stage': 'enrich',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is nested inside retained envelope persisted to RETAINED_AUDIT.
    envelope = {'source': 'pipeline', 'payload': payload}
    RETAINED_AUDIT.append(envelope)
    return 'ok'
